be/src/storage/seek_tuple.h

```diff
@@ -80,4 +82,23 @@ inline std::string SeekTuple::short_key_encode(size_t num_short_keys, uint8_t pa
     return output;
 }
 
+inline std::string SeekTuple::short_key_encode(size_t num_short_keys, std::vector<uint32_t> short_key_idxes,
+                                               uint8_t padding) const {
+    std::string output;
+    size_t n = std::min(num_short_keys, std::min(short_key_idxes.size(), _values.size()));
+    size_t val_num = _values.size();
+    for (auto i = 0; i < n; i++) {
+        if (short_key_idxes[i] >= val_num || _values[short_key_idxes[i]].is_null()) {
+            output.push_back(KEY_NULL_FIRST_MARKER);
+        } else {
+            output.push_back(KEY_NORMAL_MARKER);
+            _schema.field(short_key_idxes[i])->encode_ascending(_values[short_key_idxes[i]], &output);
+        }
+    }
+    if (_values.size() < num_short_keys) {
```

Comment:
When the `_values.size()` will be less than `num_short_keys`?

Reply:
> When the `_values.size()` will be less than `num_short_keys`?

This scenario may happened if we do partial update for the table which separate primary keys and sort keys. But we don't support partial update for this table right now. So it should never happened right now.

Reply:
So the partial update is limited due to lack of this partial update mode.

Reply:
> So the partial update is limited due to lack of this partial update mode.

Yes, table with sort key do not support partial update now.